src/phoenix/datasets/evaluators.py

```diff
@@ -62,6 +65,132 @@ def evaluate(self, example: Example, exp_run: ExperimentRun) -> EvaluationResult
+        return self._parse_eval_output(unparsed_response)
+
+    def _format_eval_template(self, experiment_run: ExperimentRun) -> str:
+        assert experiment_run.output is not None
+        result = _unwrap_json(experiment_run.output.result)
+        return self.template.format(text=str(result))
+
+    def _parse_eval_output(self, unparsed_response: str) -> EvaluationResult:
+        raw_label, explanation = (
+            _parse_label_from_explanation(unparsed_response),
+            unparsed_response,
+        )
+        label = snap_to_rail(raw_label, ["true", "false"])
+        if label == "true":
+            score = 1.0
+        elif label == "false":
+            score = 0.0
+        else:
+            raise RuntimeError(f"Could not parse LLM evaluation: {unparsed_response}")
+        return EvaluationResult(
+            score=score,
+            explanation=explanation,
+            metadata={},
+        )
+
+    @classmethod
+    def _format_base_template(cls, criteria: str, description: Optional[str] = None) -> str:
+        formatted_description = cls._description.format(criteria=criteria, description=description)
+        # leave the text field as a placeholder
+        formatted_template = cls._base_template.format(
+            criteria=criteria, description=formatted_description, text="{text}"
+        )
+        return formatted_template
+
+
+def evaluator_factory(
+    class_name: str, criteria: str, description: str
+) -> Type[ExperimentEvaluator]:
+    return type(
+        class_name,
+        (LLMCriteriaEvaluator,),
+        {
+            "__init__": lambda self, model: LLMCriteriaEvaluator.__init__(
+                self, model, criteria, description, name=class_name
+            ),
+            "__module__": __name__,
+            "name": class_name,
+            "template": LLMCriteriaEvaluator._format_base_template(criteria, description),
+        },
+    )
+
+
+LLMConcisenessEvaluator = evaluator_factory(
+    class_name="LLMConcisenessEvaluator",
+    criteria="concise",
+    description="is just a few sentences and easy to follow",
+)
+
+
+LLMHelpfulnessEvaluator = evaluator_factory(
+    class_name="LLMHelpfulnessEvaluator",
+    criteria="helpful",
+    description="provides useful information",
+)
+
+
+LLMCoherenceEvaluator = evaluator_factory(
+    class_name="LLMCoherenceEvaluator",
+    criteria="coherent",
+    description="is coherent, well-structured, and organized",
+)
+
+
+def _parse_label_from_explanation(raw_string: str) -> str:
+    label_delimiter = r"(\W*label\W*)"
+    parts = re.split(label_delimiter, raw_string, flags=re.IGNORECASE)
+    if len(parts) > 1:
+        # Find the last occurrence of the delimiter and take the part after it
+        last_index = len(parts) - 1
+        while last_index > 0:
+            if re.match(label_delimiter, parts[last_index - 1], flags=re.IGNORECASE):
+                return parts[last_index].strip()
+            last_index -= 1
+    return raw_string
```

Comment:
I am assuming the only time we hit this is when the `raw_string` does not contain "label". Do we want to fail out in this case?

Reply:
I think this flow improves on our original eval flow: if there is no "label" delimiter, it'll return the whole LLM response which we will try to snap in its entirety. If that fails we will error out and it will be captured in the annotation record